### PR TITLE
feat: adds version sub-command

### LINKED ISSUE #29: [RFE] - Add version subcommand
**What is the problem you're trying to solve**
Add a subcommand to the client that prints out the version

**Describe the solution you'd like**
As `version` subcommand added to the client

### PR DESCRIPTION
**Pull request by jpower432**
Adds version subcommand
Updates goreleaser.yaml to add ldflags
Updates Makefile to add ldflags

Closes #29

Signed-off-by: Jennifer Power <barnabei.jennifer@gmail.com>

- [X] Docs Updated
- [X] Tested added

Tested `goreleaser` changes with `goreleaser build --snapshot`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,9 +3,11 @@ GO := go
 GO_BUILD_PACKAGES := ./cmd/...
 GO_BUILD_BINDIR :=./bin
 
+GO_LD_EXTRAFLAGS := -X github.com/uor-framework/client/cli.Version="$(shell git tag | sort -V | tail -1)"
```
Comment (sabre1041) [inline Makefile:6]: Is there an interest to also include the git commit which is common i tooling in related tooling?

Comment (jpower432) [reply Makefile:6]: I can add that. That information will be useful using a build from the branch instead of the pre-built binaries.

Comment (jpower432) [reply Makefile]: Added a few useful fields like platform, go version, and build date as well.

File: Makefile [hunk 2]
```diff
@@ -2,10 +2,15 @@ GO := go
 
 GO_BUILD_PACKAGES := ./cmd/...
 GO_BUILD_BINDIR :=./bin
+GIT_COMMIT := $(or $(SOURCE_GIT_COMMIT),$(shell git rev-parse --short HEAD))
```
Comment (sabre1041) [inline Makefile:5]: If I build this, this will show the most recent tag instead of a `devel` version which would be expected

**Review by sabre1041 — CHANGES_REQUESTED** [review]
Love the new fields. 1 issue

File: Makefile (hunk 2 above)
Comment (jpower432) [reply Makefile:5]: Ack. Added a `buildData` field to add `+dev` to that version in the Makefile. WDYT?

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**